components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML2BearerGrantHandler.java

```diff
@@ -744,7 +744,9 @@ protected AuthenticatedUser buildLocalUser(OAuthTokenReqMessageContext tokReqMsg
                 (subjectIdentifier)));
 
         userTenantDomain = MultitenantUtils.getTenantDomain(subjectIdentifier);
-        if (StringUtils.isEmpty(userTenantDomain)) {
+        if (StringUtils.isEmpty(userTenantDomain) || (!serviceProvider.isSaasApp() && !subjectIdentifier.endsWith
```

Comment:
"StringUtils.isEmpty(userTenantDomain)" this check is not required since MultitenantUtils.getTenantDomain(subjectIdentifier) will return "carbon.super" if the tenantDomain is not defined in subjectIdentifier parameter